### ISSUE TITLE
Freestanding config.yaml checker

### ISSUE THREAD

**Issue by ewallace** [riboviz/riboviz#172]
Write a freestanding check script for riboviz / nextflow `config.yaml` file, to confirm that the complete file is specified correctly. Currently this information is specified in various different parts of the python code for the config parameters needed for individual steps of the workflow, but there is not a global check. Having a single file to check these means:

* nextflow configuration is checked globally before nextflow runs, instead of giving difficult-to-find errors later in the workflow.
* we can automatically check `config.yaml` files submitted to `example-datasets` repo, perhaps even with continuous integratoin.

For the latter, it needs to be *optional* to check for the presence of input fastq files. That's because fastq files will not be uploaded to example-datasets, only the config.yaml and annotation fasta/gff files.

@mikej888 is writing a version in groovy as part of the nextflow script, and suggested porting to python for the freestanding version, if it makes sense to have two distinct implementations?

**Comment by mikej888** [comment]
Nextflow workflow changes, see #167, means that the Nextflow workflow can do configuration validation without running the workflow. See https://github.com/riboviz/riboviz/blob/6f8cfb40e982b3f1b60eafc857e152fe1e742e96/docs/user/prep-riboviz-run-nextflow.md#validate-configuration.

**Comment by mikej888** [comment]
Agreed to close this ticket as the functionality is provided by the Nextflow workflow.
